main.py

```diff
@@ -258,23 +272,83 @@ def main():
     parser.add_argument('release', help="OCP release to operate on")
     parser.add_argument('--dry-run', help="Just print what would happen", action='store_true')
     parser.add_argument('--debug', '-d', help="Enable debug logging", action='store_true')
+    parser.add_argument('--filename', help="Path to file where bootimage data can be recorded; will allow for faster execution if script is run multiple times", default="deleted_images.json")
     args = parser.parse_args()
 
     if args.debug:
         logging.basicConfig(level=logging.DEBUG)
 
+    if args.filename:
+        deleted_images_filename = args.filename
 
     ### testing functions
-    #bootimages = parse_openshift_installer(args.release)
-    #print(bootimages)
-    #releases = (parse_release(args.release))
-    #images = get_images_not_tagged(releases)
+    # get aliyun images in the installer
+    bootimages = parse_openshift_installer(args.release)
+
+    # get builds with aliyun uploads from a builds.json
+    aliyun_releases = parse_release(args.release)
+
+    # preload images that should be deleted
+    deleted_images_json = {}
+    if os.path.exists(deleted_images_filename):
+        logging.debug("Found file")
+        with open(deleted_images_filename, 'r') as f:
+            deleted_images_json = json.load(f)
+
+    # find the builds from builds.json that are not in bootimages
+    for buildid in aliyun_releases.keys():
+        if buildid in bootimages:
+            print(f"Build ID {buildid} in bootimage metadata; tagging with bootimage=true")
+            for region in aliyun_releases[buildid]:
+                # TODO: uncomment this when we want to go live
+                #tag_image(region, image_id, tag_key="bootimage", tag_value="true")
```

Comment:
I was looking it again, and maybe we could have a list first for the image that are not tagged at all.